tools/harness-simulation/harness/Thread_Harness/THCI/OpenThread_Sim.py

```diff
@@ -71,26 +78,37 @@ def __connect(self, device, node_id):
             else:
                 raise Exception('Password error')
 
-        self.__stdin, self.__stdout, _ = self.__handle.exec_command(device + ' ' + str(node_id), get_pty=True)
-        self.__stdout.channel.setblocking(0)
+        # Avoid SSH connection lost after inactivity for a while
+        self.__handle.get_transport().set_keepalive(self.KEEPALIVE_INTERVAL)
 
-        # Wait some time for initiation
-        time.sleep(0.1)
+        self.__stdin, self.__stdout, _ = self.__handle.exec_command(device + ' ' + str(node_id))
 
-    @watched
-    def close(self):
+        # Receive the output in blocking mode with a timeout
+        # Setting the timeout is for robustness in case of being blocked by unknown errors
+        self.__stdout.channel.settimeout(self.timeout)
```

Comment:
Yes, it does. There is a statement`return self.__stdout.readline().rstrip()` in the method. Every read method called by `self.__stdout` will be affected with the timeout.

Reply:
Originally, we expect `_cliReadLine()` to return immediately when there is no data to read.
This design is used because:
- When a device is `factoryreset`, it does not output `Done`, and there might be residual data in the stdout that needs to be cleaned. Setting a timeout will harm the device responsiveness in such case.

So, I think we should follow the protocol:
- _cliReadLine should return one next line from the device
- If the line is not available yet (e.g. no data, or line end not read yet), return None

Reply:
Referencing the implementation of class OpenThreadTHCI in *OpenThread.py*, Harness calls `reset()` by executing `self.__sendCommand('factoryreset', expectEcho=False)`, which does not run `_cliReadLine()` after sending the `factoryreset` command.

In the case that there is residual data, it cannot be the output of any commands executed before in the current node as far as I know. It will never execute the next command when there is still more output by the previous commands. The only possibility is the output of asynchronous messages, such as receiving the data sent by other nodes using `udp send`, and other nodes joining the commissioner. OpenThreadTHCI can filter and ignore the asynchronous messages properly. It may read the residual data at the next startup and ignores it without bringing any trouble. Moreover, I think it will not harm the device responsiveness even there are errors, since the asynchronous messages are always valid and can be distinguished while synchronous messages have an ending of several fixed patterns.

In fact, I think the non-blocking `_cliReadLine()` before may be a redundant design. Because it never returns `None` in the blocking mode with a timeout during all the test cases conducted afterwards.

Reply:
Change to non-blocking mode for consistency with the assumption about the behavior of `_cliReadLine()`.